Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -86,13 +84,6 @@ internal class OperationExecutor {
         }
 
         if let session = session {
-            if case .ended = session.state {
```

Comment:
this logic is already handled in `session.getConnection`. I think it's overly defensive to check in both places. the only upside of checking here is that we find out a little sooner. but these are both edge cases/user error so I'm not sure it's worth optimizing for them.